pennylane_lightning_gpu/src/tests/mpi/Test_MPI_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -0,0 +1,333 @@
+                                                nGlobalIndexBits,              \
+                                                nLocalIndexBits);              \
+                sv.CopyHostDataToGpu(local_state, false);                      \
+                sv.GATE_METHOD(WIRE, false);                                   \
+                sv.CopyGpuDataToHost(local_state.data(),                       \
+                                     static_cast<std::size_t>(subSvLength));   \
+                                                                               \
+                SVDataGPU<TestType> svdat{(NUM_QUBITS), init_sv};              \
+                if (mpi_manager.getRank() == 0) {                              \
+                    svdat.cuda_sv.GATE_METHOD(WIRE, false);                    \
+                    svdat.cuda_sv.CopyGpuDataToHost(expected_sv.data(),        \
+                                                    svLength);                 \
+                }                                                              \
+                auto expected_local_sv = mpi_manager.scatter(expected_sv, 0);  \
+                CHECK(local_state == Pennylane::approx(expected_local_sv));    \
+            }                                                                  \
+        }                                                                      \
+        SECTION("Apply using dispatcher") {                                    \
+            SECTION("Operation on target wire") {                              \
+                StateVectorCudaMPI<TestType> sv(mpi_manager, mpi_buffersize,   \
+                                                nGlobalIndexBits,              \
+                                                nLocalIndexBits);              \
+                sv.CopyHostDataToGpu(local_state, false);                      \
+                sv.applyOperation(GATE_NAME, WIRE, false);                     \
+                sv.CopyGpuDataToHost(local_state.data(),                       \
+                                     static_cast<std::size_t>(subSvLength));   \
+                SVDataGPU<TestType> svdat{(NUM_QUBITS), init_sv};              \
+                if (mpi_manager.getRank() == 0) {                              \
+                    svdat.cuda_sv.applyOperation(GATE_NAME, WIRE, false);      \
+                    svdat.cuda_sv.CopyGpuDataToHost(expected_sv.data(),        \
+                                                    svLength);                 \
+                }                                                              \
+                auto expected_local_sv = mpi_manager.scatter(expected_sv, 0);  \
+                CHECK(local_state == Pennylane::approx(expected_local_sv));    \
+            }                                                                  \
+        }                                                                      \
+    }
```

Comment:
Do you have a special reason to define this function with a macro?
A usual function would in principle work with CATCH2.

Reply:
Code maintenance will be easier with macros. As you might noticed, the total number of lines is reduced from over 4000 lines to around 300 lines.

Reply:
Sorry, but I don't think I understood why. 
You could have all you have inside this macro function inside a usual function, and this would reduce code in the same way.
Is there something here that would make it difficult to reuse a usual function?
It is better to avoid the use of preprocessor directives or macros everywhere we can.

Reply:
Hi @AmintorDusko , I thought I have already answered your question on why I used macro. In fact, I tried to put everything into a usual function and call it before. But it didn't work for the `sv.applyxxxgate` methods and wouldn't save thousands of lines like using macro here.